demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1354 @@
+
+/**
+ * @brief Decode memory.
+ */
+uint8_t decodeMem[ otaconfigFILE_BLOCK_SIZE ];
+
+/**
+ * @brief Bitmap memory.
+ */
+uint8_t bitmap[ OTA_MAX_BLOCK_BITMAP_SIZE ];
+
+/**
+ * @brief Event buffer.
+ */
+static OtaEventData_t eventBuffer[ otaconfigMAX_NUM_OTA_DATA_BUFFERS ];
+
+/**
+ * @brief The buffer passed to the OTA Agent from application while initializing.
+ */
+static OtaAppBuffer_t otaBuffer =
+{
+    .pUpdateFilePath    = updateFilePath,
+    .updateFilePathsize = OTA_MAX_FILE_PATH_SIZE,
+    .pCertFilePath      = certFilePath,
+    .certFilePathSize   = OTA_MAX_FILE_PATH_SIZE,
+    .pStreamName        = streamName,
+    .streamNameSize     = OTA_MAX_STREAM_NAME_SIZE,
+    .pDecodeMemory      = decodeMem,
+    .decodeMemorySize   = otaconfigFILE_BLOCK_SIZE,
+    .pFileBitmap        = bitmap,
+    .fileBitmapSize     = OTA_MAX_BLOCK_BITMAP_SIZE
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Sends an MQTT CONNECT packet over the already connected TCP socket.
+ *
+ * @param[in] pMqttContext MQTT context pointer.
+ * @param[in] createCleanSession Creates a new MQTT session if true.
+ * If false, tries to establish the existing session if there was session
+ * already present in broker.
+ * @param[out] pSessionPresent Session was already present in the broker or not.
+ * Session present response is obtained from the CONNACK from broker.
+ *
+ * @return EXIT_SUCCESS if an MQTT session is established;
+ * EXIT_FAILURE otherwise.
+ */
+static int establishMqttSession( MQTTContext_t * pMqttContext );
+
+/**
+ * @brief Close an MQTT session by sending MQTT DISCONNECT.
+ *
+ * @param[in] pMqttContext MQTT context pointer.
+ *
+ * @return EXIT_SUCCESS if DISCONNECT was successfully sent;
+ * EXIT_FAILURE otherwise.
+ */
+static int disconnectMqttSession( MQTTContext_t * pMqttContext );
+
+/**
+ * @brief Publish message to a topic.
+ *
+ * This function publishes a message to a given topic & QoS.
+ *
+ * @param[in] pacTopic Mqtt topic filter.
+ *
+ * @param[in] topicLen Length of the topic filter.
+ *
+ * @param[in] pMsg Message to publish.
+ *
+ * @param[in] msgSize Message size.
+ *
+ * @param[in] qos Quality of Service
+ *
+ * @return OtaMqttSuccess if success , other error code on failure.
+ */
+static OtaMqttStatus_t mqttPublish( const char * const pacTopic,
+                                    uint16_t topicLen,
+                                    const char * pMsg,
+                                    uint32_t msgSize,
+                                    uint8_t qos );
+
+/**
+ * @brief Subscribe to the Mqtt topics.
+ *
+ * This function subscribes to the Mqtt topics with the Quality of service
+ * received as parameter. This function also registers a callback for the
+ * topicfilter.
+ *
+ * @param[in] pTopicFilter Mqtt topic filter.
+ *
+ * @param[in] topicFilterLength Length of the topic filter.
+ *
+ * @param[in] qos Quality of Service
+ *
+ * @param[in] callback Callback to be registered.
```

Comment:
```suggestion
 * @param[in] callback Callback to be registered for handling an incoming PUBLISH message on the topic.
```

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/726a7d066006ec340f26308d37075a414069f56b